### PR TITLE
Add task solution

### PR DESCRIPTION
**Pull request by SerhiiUnhurian**

### CODE REVIEW

File: src/groupByKey.ts [hunk 1]
```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
...
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyValue = item[key] as string;
+
+    if (!result[keyValue]) {
+      result[keyValue] = [item];
+    } else {
+      result[keyValue].push(item);
+    }
```
Comment (VitaliyBodnar) [inline src/groupByKey.ts:15]: make sense to use ternary operator here as it will make a code shorter

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]

File: src/groupByKey.ts (hunk 1 above)
Comment (SerhiiUnhurian) [reply src/groupByKey.ts:15]: I think this code is quite short
There will be even more code to fix

File: src/groupByKey.ts [hunk 2]
```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
...
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyValue = item[key] ? (item[key] as string) : '';
+
+    return result[keyValue] ? result[keyValue].push(item)
+      : (result[keyValue] = [item]);
+  });
```
Comment (VitaliyBodnar) [inline src/groupByKey.ts:13]:
```suggestion
    return result[keyValue] 
      ? result[keyValue].push(item)
      : result[keyValue] = [item];
  });
```

File: src/groupByKey.ts [hunk 3]
```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const keyValue = item[key] ? (item[key] as string) : '';
```
Comment (VitaliyBodnar) [inline src/groupByKey.ts:9]:
```suggestion
    const keyValue = item[key] || '';
```

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]

File: src/groupByKey.ts (hunk 3 above)
Comment (SerhiiUnhurian) [reply src/groupByKey.ts:9]: we have a problem here 
Screenshot 2023-11-14 at 15 08 56
Screenshot 2023-11-14 at 15 08 49

Comment (VitaliyBodnar) [reply src/groupByKey.ts:9]: okay, let's try this one 
`const keyValue = item[key] as string || ''`;

**Review by VitaliyBodnar — APPROVED** [review]

**Pull request closed without merging**